core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/stacks/instancegroup/InstanceGroupV4RequestToHostGroupConverter.java

```diff
@@ -35,14 +35,14 @@
 
     @Override
     public HostGroup convert(InstanceGroupV4Request source) {
-        CloudbreakUser cloudbreakUser = restRequestThreadLocalService.getCloudbreakUser();
-        User user = userService.getOrCreate(cloudbreakUser);
-        Workspace workspace = workspaceService.get(restRequestThreadLocalService.getRequestedWorkspaceId(), user);
         HostGroup hostGroup = new HostGroup();
-        hostGroup.setName(source.getName());
+        hostGroup.setName(source.getName().toLowerCase());
         hostGroup.setRecoveryMode(source.getRecoveryMode());
         Set<String> recipeNames = source.getRecipeNames();
         if (!CollectionUtils.isEmpty(recipeNames)) {
+            CloudbreakUser cloudbreakUser = restRequestThreadLocalService.getCloudbreakUser();
```

Comment:
will this be slow in request time?

Reply:
Sorry, but I don't understand the question, what will be slow here?

Reply:
User user = userService.getOrCreate(cloudbreakUser);

Reply:
I just moved into the if, because we fetched the recipes even if we didn't send the names in the request